### PR TITLE
[example] Add example for custom error for exceeded usage limits

### PR DESCRIPTION
**Pull request by mayorova**
This PR provides an example of how to return a different error message to the client, when the call is not authorized, because usage limits are exceeded for the application.

Basically, it's adaptation of this [CodeHub snippet](https://support.3scale.net/codehub/custom-error-codes) to new APIcast.

It can be used as a workaround until https://github.com/3scale/apicast/issues/368 is closed.

### CODE REVIEW

File: examples/custom-module/usage_limits_error.lua [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+proxy.handle_backend_response = function(self, cached_key, response, ttl)
+  ngx.log(ngx.DEBUG, '[backend] response status: ', response.status, ' body: ', response.body)
+
+  local authorized, reason = self.cache_handler(self.cache, cached_key, response, ttl)
+
+  if not authorized then
+    local usage_limits_exceeded = utils.match_xml_element(response.body, 'reason', backend_reason)
+    if usage_limits_exceeded then
+      error_limits_exceeded(cached_key)
```
Comment (mikz) [inline examples/custom-module/usage_limits_error.lua]: I guess this fails when it not the first request?

Comment (mayorova) [reply examples/custom-module/usage_limits_error.lua]: It doesn't complain, but indeed, we shouldn't be trying to send response in post_action, I've fixed this.

When authorization fails, all consequent calls will have to renew auth synchronously (until getting a `200`), so this should work always.

File: examples/custom-module/usage_limits_error.lua [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  ngx.log(ngx.DEBUG, '[backend] response status: ', response.status, ' body: ', response.body)
+
+  local authorized, reason = self.cache_handler(self.cache, cached_key, response, ttl)
+  local phase = ngx.get_phase()
+
+  -- only respond to the client with the usage limits error in 'access' phase, ignore in 'post_action'
+  if not authorized and phase == 'access' then
+    local usage_limits_exceeded = utils.match_xml_element(response.body, 'reason', backend_reason)
+    if usage_limits_exceeded then
+      error_limits_exceeded(cached_key)
```
Comment (mikz) [inline examples/custom-module/usage_limits_error.lua]: would be nicer to have there explicit `return` to show it is going to end the flow.

Comment (mayorova) [reply examples/custom-module/usage_limits_error.lua]: @mikz `return error_limits_exceeded(cached_key)` ?

Comment (mikz) [reply examples/custom-module/usage_limits_error.lua]: yep.

Comment (mayorova) [reply examples/custom-module/usage_limits_error.lua]: done :)

**Pull request merged**